### PR TITLE
Deprecate method #redirect_back_or_default

### PR DESCRIPTION
**Pull request by cpfergus1**
## Summary

Rails introduced [redirect_back](https://api.rubyonrails.org/classes/ActionController/Redirecting.html#method-i-redirect_back) in rails 5+ and [redirect_back_or_to](https://api.rubyonrails.org/classes/ActionController/Redirecting.html#method-i-redirect_back_or_to) in rails 7+. The naming of Solidus method #redirect_back_or_default lead to some confusion between the two methods and it appeared that it could be replaced by the included Rails methods to eliminate redundancy and simplify the code. After further investigation, the two functions create different results and cannot be used interchangeably. Devise, however, already contains a similar function to #redirect_back_or_default, so the functionality would be best moved to solidus_auth_devise.

This will affect the following gems

solidus_frontend - https://github.com/solidusio/solidus_frontend/pull/9
solidus_starter_frontend - https://github.com/solidusio/solidus_starter_frontend/pull/241
solidus_auth_devise - https://github.com/solidusio/solidus_auth_devise/pull/228
solidus_social - https://github.com/solidusio-contrib/solidus_social/pull/109
solidus_active_shipping - archived, No plans to update
solidus_paypal_marketplace - https://github.com/solidusio-contrib/solidus_paypal_marketplace/pull/103

## Checklist

Check out our [PR guidelines](https://github.com/solidusio/solidus/blob/master/CONTRIBUTING.md#pull-request-guidelines) for more details.

The following are mandatory for all PRs:

- [x] I have written a thorough PR description.
- [x] I have kept my commits small and atomic.
- [x] I have used clear, explanatory commit messages.

The following are not always needed (~cross them out~ if they are not):

- [x] I have added automated tests to cover my changes.
- [ ] ~~I have attached screenshots to demo visual changes.~~
- [ ] ~~I have opened a PR to update the [guides](https://github.com/solidusio/edgeguides).~~
- [ ] ~~I have updated the readme to account for my changes.~~

### CODE REVIEW

File: core/lib/spree/core/controller_helpers/auth.rb [hunk 1]
```diff
@@ -36,11 +36,19 @@ def current_ability
           @current_ability ||= Spree::Ability.new(spree_current_user)
         end
 
-        def redirect_back_or_default(default)
-          redirect_to(session["spree_user_return_to"] || default)
+        def stored_location_or(fallback_location)
```
Comment (waiting-for-dev) [inline core/lib/spree/core/controller_helpers/auth.rb:39]: We should probably implement it in `solidus_auth_devise`

**Review by waiting-for-dev — CHANGES_REQUESTED** [review]

**Review by waiting-for-dev — APPROVED** [review]
Thanks, @cpfergus1! Putting it on hold until the PRs on the extensions are ready.

**Review by jarednorman — APPROVED** [review]

File: core/app/models/spree/user_last_url_storer.rb [hunk 2]
```diff
@@ -26,6 +26,7 @@ def self.rules
     #  or its subclasses. The controller will be passed to each rule for matching.
     def initialize(controller)
       @controller = controller
+      Spree::Deprecation.warn("This class will be removed without replacement on the realease of Solidus 4.0")
```
Comment (kennyadsl) [inline core/app/models/spree/user_last_url_storer.rb:29]: But are we still using this in core [here](https://github.com/solidusio/solidus/blob/a22723079f88387f53f72b503db875b9d97aa3f5/core/lib/spree/core/controller_helpers/auth.rb#L54)? If it's not called anymore, that method should be deprecated as well.

**Review by kennyadsl — COMMENTED** [review]
Thanks Connor, I just left a question.

File: core/app/models/spree/user_last_url_storer.rb (hunk 2 above)
Comment (cpfergus1) [reply core/app/models/spree/user_last_url_storer.rb:29]: Good point!
I also removed that method from solidus_starter_frontend and solidus_auth_devise!

Comment (kennyadsl) [reply core/app/models/spree/user_last_url_storer.rb]: Thanks!

**Review by kennyadsl — APPROVED** [review]
Thanks Connor, excellent work!

**Pull request merged**